packages/firestore/src/core/transaction.ts

```diff
@@ -64,10 +64,11 @@ export class Transaction {
     this.ensureCommitNotCalled();
 
     if (this.mutations.length > 0) {
-      throw new FirestoreError(
+      this.lastTransactionError = new FirestoreError(
```

Comment:
It is possible that the same error gets thrown twice: once is here when we detect it, and throw again when we call commit() since it is stored into this.lastTransactionError.